pages/api/session.js

```diff
@@ -44,40 +47,60 @@ async function login (req, res) {
       }
 
     const { session_token, user: {firstName, lastName, roles, capabilities, defaultEntityType, forcePasswordChange} } = await People.login(email, password);
-    req.session.session_token = session_token;
-    req.session.cache = {
-      firstName,
-      lastName,
-      roles,
-      capabilities,
-      defaultEntityType,
-      homepage: '/admin',
-      forcePasswordChange,
-    };
-
-    const token = generateToken({ firstName, lastName });
-    setCookie('token', token, { req, res, httpOnly: true, secure: true, sameSite: 'none' });
-
-    await req.session.save();    
-    res.status(OK).json({ forcePasswordChange });
-  
-  } catch (error) {
-    if (error instanceof UnauthorizedError) {
-      res.status(UNPROCESSABLE_ENTITY).json({message: "Invalid username or password."});
-      return;
-    } else {
-      await defaultErrorHandler(error, req, res);
+
+    const { origin, host } = req.headers;
+    const isFromCMS = origin.endsWith(host);
+
+    const cmsUsers = "CMS Users"
+    const isCMSUser = roles.includes(cmsUsers);
+
+    if (isFromCMS && !isCMSUser) throw new UnauthorizedError();
+
+    // We need the session token to identify the user when they are 
+    // updating their user info. Another option is to take the ID of the user in 
+    // the database. The session token is stored in two places if we are logged in 
+    // to the CMS (JWT and req.session). Ideally it should only be placed in one, 
+    // but that would require major code changes.
+    const token = generateToken({ firstName, lastName, email, sessionToken: session_token });
+    const response = { message: 'Sucessfully logged in!' };
+
+    // If a user logged in from the CMS, then we need to store the data from CMS.
+    // Otherwise, the data is just extra baggage for the 'external website'.
+    if (isFromCMS) {
+        req.session.session_token = session_token;
+        req.session.cache = {
+          token,
+          firstName,
+          lastName,
+          roles,
```

Comment:
`roles` is deprecated. We do not base the ability of the user based on roles, but on the capability attached on that role. `roles` can be deleted as far as I can tell.